metadata-service/restli-servlet-impl/src/main/java/com/linkedin/metadata/resources/entity/AspectResource.java

```diff
@@ -153,15 +155,18 @@ public Task<String> ingestProposal(
     String actorUrnStr = authentication.getActor().toUrnStr();
     final AuditStamp auditStamp = new AuditStamp().setTime(_clock.millis()).setActor(Urn.createFromString(actorUrnStr));
 
-    final List<MetadataChangeProposal> additionalChanges =
-        AspectUtils.getAdditionalChanges(metadataChangeProposal, _entityService);
-
     return RestliUtil.toTask(() -> {
       log.debug("Proposal: {}", metadataChangeProposal);
       try {
-        Urn urn = _entityService.ingestProposal(metadataChangeProposal, auditStamp, asyncBool).getUrn();
-        additionalChanges.forEach(proposal -> _entityService.ingestProposal(proposal, auditStamp, asyncBool));
-        tryIndexRunId(urn, metadataChangeProposal.getSystemMetadata(), _entitySearchService);
+        EntityService.IngestProposalResult result = _entityService.ingestProposal(metadataChangeProposal, auditStamp, asyncBool);
+        Urn urn = result.getUrn();
+
+        AspectUtils.getAdditionalChanges(metadataChangeProposal, _entityService)
+                .forEach(proposal -> _entityService.ingestProposal(proposal, auditStamp, asyncBool));
+
+        if (!result.isDidUpdate()) {
```

Comment:
Why is this check required prior to attempting to index the run id? We should always index the latest run ID so we can show which entities were touched by which run (in the Ingestion Run Summary)